### PR TITLE
add round trip equality check to fuzz tests

### PR DESCRIPTION
**Pull request by sukunrt**

### CODE REVIEW

**Review by Jorropo — APPROVED** [review]

File: multiaddr_test.go [hunk 1]
```diff
@@ -848,3 +870,19 @@ func BenchmarkUniqueAddrs(b *testing.B) {
...
 }
+
+func TestDNS(t *testing.T) {
+	b := []byte("7*000000000000000000000000000000000000000000")
+	a, err := NewMultiaddrBytes(b)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println(b)
+	fmt.Println([]byte(a.String()))
```
Comment (Jorropo) [inline multiaddr_test.go:881]:
```suggestion
```

File: multiaddr_test.go [hunk 2]
```diff
@@ -848,3 +870,19 @@ func BenchmarkUniqueAddrs(b *testing.B) {
...
+	aa := StringCast(a.String())
+	if !a.Equal(aa) {
+		fmt.Println(a.Bytes())
+		fmt.Println(aa.Bytes())
```
Comment (Jorropo) [inline multiaddr_test.go:885]:
```suggestion
```

**Review by Jorropo — APPROVED** [review]

File: multiaddr_test.go (hunk 2 above)
Comment (sukunrt) [reply multiaddr_test.go]: sorry about these 😓

Comment (Jorropo) [reply multiaddr_test.go]: it's fine

**Pull request merged**